spec.html

```diff
@@ -7264,10 +7277,10 @@ <h1>GetIdentifierReference ( _env_, _name_, _strict_ )</h1>
         <p>The abstract operation GetIdentifierReference takes arguments _env_ (an Environment Record or *null*), _name_ (a String), and _strict_ (a Boolean). It performs the following steps when called:</p>
         <emu-alg>
           1. If _env_ is the value *null*, then
-            1. Return a value of type Reference whose base value component is *undefined*, whose referenced name component is _name_, and whose strict reference flag is _strict_.
+            1. Return Reference { [[ReferencedName]]: _name_, [[Strict]]: _strict_ }.
```

Comment:
All fields of the Reference should be present whenever one is constructed. It's not clear here whether `[[Base]]` should have the value `*undefined*` (as it does in the current text), `empty`, or something else.

Reply:
+1, here and below, all fields should be explicitly filled in.

Could make convenience AOs like MakeReference and MakeSuperReference, etc, but it doesn't look like there are that many uses, so spelling out the full Record each time also seems fine to me.

Reply:
I suspect, though, that that's the only Record for which all fields are not always present.

Reply:
With ee73d98 and 5a03a3a, Reference records are always defined with all fields present.